src/aiida/common/hashing.py

```diff
@@ -196,27 +197,29 @@ def _(mapping, **kwargs):
 
 
 @_make_hash.register(numbers.Real)
-def _(val, **kwargs):
+def _(val: numbers.Real, **kwargs) -> list[bytes]:
     """Before hashing a float, convert to a string (via rounding) and with a fixed number of digits after the comma.
     Note that the `_single_digest` requires a bytes object so we need to encode the utf-8 string first
     """
     return [_single_digest('float', float_to_text(val, sig=AIIDA_FLOAT_PRECISION).encode('utf-8'))]
 
 
 @_make_hash.register(Decimal)
-def _(val, **kwargs):
+def _(val: Decimal, **kwargs) -> list[bytes]:
     """While a decimal can be converted exactly to a string which captures all characteristics of the underlying
     implementation, we also need compatibility with "equal" representations as int or float. Hence we are checking
     for the exponent (which is negative if there is a fractional component, 0 otherwise) and get the same hash
     as for a corresponding float or int.
     """
-    if val.as_tuple().exponent < 0:
+    # TODO: This is a possible problem!
+    # Unsupported operand types for < ("Literal['n']" and "int")
+    if val.as_tuple().exponent < 0:  # type: ignore[operator]
```

Comment:
It would be better to explicitly deal with it I guess, but I don't think this will currently happen. The hashing is done after all attributes are validated, and that already checks for `inf/NaN` values, which are not supported:
```python
In [1]: from decimal import Decimal

In [2]: data = Data()

In [3]: data.base.attributes.set('decimal', Decimal('NaN'))

In [4]: data.store()
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
Cell In[4], line 1
----> 1 data.store()

ValidationError: nan and inf/-inf can not be serialized to the database

In [5]: data.base.caching.compute_hash()
---------------------------------------------------------------------------
InvalidOperation                          Traceback (most recent call last)
Cell In[5], line 1
----> 1 data.base.caching.compute_hash()
                                                                                                                                                                                                                 File ~/code/aiida/env/dev/aiida-core/src/aiida/orm/nodes/caching.py:36, in NodeCaching.compute_hash(self, ignore_errors, **kwargs)
     31 """Return the computed hash for this node based on its attributes.
     32
     33 :param ignore_errors: return ``None`` on ``aiida.common.exceptions.HashingError`` (logging the exception)
     34 """
     35 if not self._node.is_stored:
---> 36     raise exceptions.InvalidOperation('You can get the hash only after having stored the node')
     38 return self._compute_hash(ignore_errors=ignore_errors, **kwargs)
                                                                                                                                                                                                                 InvalidOperation: You can get the hash only after having stored the node
```

And since hashing (through public API) can only be done after storing, this should be ok.